src/rust/engine/fs/src/lib.rs

```diff
@@ -377,7 +381,14 @@ impl PosixFS {
       if file_type.is_dir() {
         stats.push(Stat::Dir(Dir(path)));
       } else if file_type.is_file() {
-        stats.push(Stat::File(File(path)));
+        let is_executable = std::fs::metadata(dir_abs.join(dir_entry.file_name()))?
```

Comment:
The `read_dir` call "gives you" (depending on which syscalls happen to be available) the path metadata: https://doc.rust-lang.org/std/fs/struct.DirEntry.html#method.metadata

Reply:
Done.